src/spoke/libraries/UserPositionDebt.sol

```diff
@@ -0,0 +1,112 @@
+        premiumOffsetRay: userPosition.premiumOffsetRay,
+        drawnIndex: drawnIndex
+      });
+  }
+
+  /// @notice Calculates the premium delta for a user position given a new risk premium.
+  /// @param userPosition The user position.
+  /// @param drawnIndex The current drawn index.
+  /// @param riskPremium The new risk premium, expressed in BPS.
+  /// @param restoredPremiumRay The amount of premium to be restored, expressed in asset units and scaled by RAY.
+  /// @return The calculated premium delta.
+  function getPremiumDelta(
+    ISpoke.UserPosition storage userPosition,
+    uint256 drawnIndex,
+    uint256 riskPremium,
+    uint256 restoredPremiumRay
+  ) internal view returns (IHubBase.PremiumDelta memory) {
+    uint256 oldPremiumShares = userPosition.premiumShares;
+    int256 oldPremiumOffsetRay = userPosition.premiumOffsetRay;
+    uint256 premiumDebtRay = Premium.calculatePremiumRay({
+      premiumShares: oldPremiumShares,
+      premiumOffsetRay: oldPremiumOffsetRay,
+      drawnIndex: drawnIndex
+    });
+
+    uint256 newPremiumShares = userPosition.drawnShares.percentMulUp(riskPremium);
+    int256 newPremiumOffsetRay = (newPremiumShares * drawnIndex).signedSub(
+      premiumDebtRay - restoredPremiumRay
+    );
+
+    return
+      IHubBase.PremiumDelta({
+        sharesDelta: newPremiumShares.signedSub(oldPremiumShares),
+        offsetRayDelta: newPremiumOffsetRay - oldPremiumOffsetRay,
+        restoredPremiumRay: restoredPremiumRay
+      });
+  }
+
+  /// @notice Applies the premium delta to the user position.
+  /// @param userPosition The user position.
+  /// @param premiumDelta The premium delta to apply.
+  function applyPremiumDelta(
+    ISpoke.UserPosition storage userPosition,
+    IHubBase.PremiumDelta memory premiumDelta
+  ) internal {
+    userPosition.premiumShares = userPosition
+      .premiumShares
+      .add(premiumDelta.sharesDelta)
+      .toUint120();
+    userPosition.premiumOffsetRay = (userPosition.premiumOffsetRay + premiumDelta.offsetRayDelta)
+      .toInt200();
+  }
+
+  /// @return The user's drawn debt.
+  /// @return The user's premium debt.
+  /// @return The user's premium debt, expressed in asset units and scaled by RAY.
+  function getDebt(
+    ISpoke.UserPosition storage userPosition,
+    IHubBase hub,
+    uint256 assetId
+  ) internal view returns (uint256, uint256, uint256) {
+    return userPosition.getDebt(hub.getAssetDrawnIndex(assetId));
+  }
+
+  /// @return The user's drawn debt.
+  /// @return The user's premium debt.
+  /// @return The user's premium debt, expressed in asset units and scaled by RAY.
+  function getDebt(
+    ISpoke.UserPosition storage userPosition,
+    uint256 drawnIndex
+  ) internal view returns (uint256, uint256, uint256) {
+    uint256 premiumDebtRay = userPosition.calculatePremiumRay(drawnIndex);
+    return (
+      userPosition.drawnShares.rayMulUp(drawnIndex),
+      premiumDebtRay.fromRayUp(),
+      premiumDebtRay
```

Comment:
wondering if we should return `premiumDebtRay` only

Reply:
[8e6ea16](https://github.com/aave/aave-v4/pull/1019/commits/8e6ea169d76ff0e51ccae3c8a48bf8fe060ba417)